src/backend/distributed/commands/utility_hook.c

```diff
@@ -513,6 +514,26 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 		CommandCounterIncrement();
 
 		PostStandardProcessUtility(parsetree);
+
+		if (IsA(parsetree, VacuumStmt))
+		{
+			/*
+			 * We commit the current transaction here so that the global lock
+			 * taken from the shell table for VACUUM is released, which would block execution
+			 * of shard placements.
+			 */
+			if (ActiveSnapshotSet())
+			{
+				PopActiveSnapshot();
```

Comment:
This line should probably change, not sure what happens if we don't pop active snapshots other than getting a WARNING, should look into pg codebase a bit

Reply:
Any reason why we to do this here, rather than in PostprocessVacuumStmt?

Reply:
No I moved them